usecases/auth/authorization/rbac/authorizer.go

```diff
@@ -51,19 +59,24 @@ func (m *manager) authorize(principal *models.Principal, verb string, skipAudit
 			return err
 		}
 
-		logger.WithFields(logrus.Fields{
-			"resources": prettyPermissionsResources(perm),
-			"results":   prettyStatus(allowed),
+		// Store the results for this iteration
+		permResults = append(permResults, logrus.Fields{
+			"resource": prettyPermissionsResources(perm),
+			"results":  prettyStatus(allowed),
 		})
 
-		if !skipAudit {
-			logger.Info()
-		}
-
 		if !allowed {
+			// Log all results before returning error
+			logger.WithField("permissions", permResults).Error("authorization denied")
```

Comment:
should we skip logging if we have `skipAudit`? Otherwise this would show up in the role-scope checking